### ISSUE TITLE
installation.md 'From Source' instructions fail: cargo install --path . errors on the virtual workspace manifest

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#424]
**Priority:** high

## Current state

`docs/src/installation.md` "From Source" section says:

```bash
git clone https://github.com/joshrotenberg/mdbook-lint.git
cd mdbook-lint
cargo install --path .
```

Running this exactly as written from the repo root fails, because the root `Cargo.toml` is a virtual workspace manifest (`[workspace]` only, no `[package]`) — the installable binary crate is `crates/mdbook-lint-cli`. Confirmed:

```text
$ cargo install --path .
error: found a virtual manifest at `/.../mdbook-lint/Cargo.toml` instead of a package manifest
```

Separately, the "Verification" section (lines 66-70) of the same page shows:

```
mdbook-lint --version
```

> You should see output similar to:
>
> ```
> mdbook-lint 0.1.0
> ```

The current published version is 0.14.3 (per `Cargo.toml`'s `workspace.package.version`). `0.1.0` is a leftover from the project's earliest release and has not been updated across roughly 14 minor version bumps, which will confuse a new user into thinking something is badly wrong with their install.

## Desired state

- "From Source" instructions use the correct path: `cargo install --path crates/mdbook-lint-cli`.
- The sample verification output shows a realistic current version, or better, avoids hardcoding a specific version at all (e.g. "output will show the installed version, e.g. `mdbook-lint x.y.z`").

## Code example

```bash
# Fails today:
cargo install --path .

# Works:
cargo install --path crates/mdbook-lint-cli
```

## Implementation notes

- `docs/src/installation.md` lines 42-44 (path fix) and lines 66-70 (version fix).
- Since the version drifts every release, consider replacing the literal `mdbook-lint 0.1.0` block with a generic placeholder rather than a real version string, to avoid this going stale again.

**Comment by joshrotenberg** [comment]
ready: docs/src/installation.md: change "From Source" to `cargo install --path crates/mdbook-lint-cli`, and replace the hardcoded `mdbook-lint 0.1.0` verification sample with a generic version placeholder so it does not go stale each release.
